app/agents/base/BaseAgent.py

```diff
@@ -46,3 +52,28 @@ async def stop(self):
 
     async def setup(self):
         self.logger.info("setup - started")
+
+    async def prepare_message(
+        self,
+        to_jid: JID,
+        performative: str,
+        ontology: str,
+        type: MessageType,
```

Comment:
Ta metoda wydaje się być przydatna i dobrze żeby została w `BaseAgent`, ale ten `MessageType` i jego logika tyczy się już tylko komunikacji między agentami stage i recruitment. Tą customową logikę należało by wyodrębnić i przeniść do "zainteresowanych" agentów

Reply:
Albo lepiej - można zostawić jak jest bo to dobry mechanizm, ale może by zrobić też customową metodę receive w której trzeba podać typ wiadomości na którą ma waitować?